oprf/keys.go

```diff
@@ -44,19 +47,24 @@ func (k *PrivateKey) Public() *PublicKey {
 }
 
 // GenerateKey generates a pair of keys in accordance with the suite.
-func GenerateKey(id SuiteID) (*PrivateKey, error) {
+func GenerateKey(id SuiteID, rnd io.Reader) (*PrivateKey, error) {
 	suite, err := suiteFromID(id, BaseMode)
 	if err != nil {
 		return nil, err
 	}
-	return suite.generateKey(), nil
+	if rnd == nil {
```

Comment:
Why can't the caller pass in rand.Reader themselves?

Reply:
now, it panics. please review changes.

Reply:
I think you might not even need the if and let the read function panic.

Reply:
Here we are following the api pattern in go stdlib.
e.g. https://golang.org/pkg/crypto/ecdsa/#GenerateKey